plugins/usbdmx/DMXCProjectsNodleU1Factory.cpp

```diff
@@ -33,22 +33,53 @@ namespace usbdmx {
 const uint16_t DMXCProjectsNodleU1Factory::VENDOR_ID = 0x16d0;
 const uint16_t DMXCProjectsNodleU1Factory::PRODUCT_ID = 0x0830;
 
+// Those IDs are not officially registered or "free for everyone"
+// but there seems to be one clone using them.
+// See also: https://github.com/mcallegari/qlcplus/blob/3b69452d0679333c6e60ff0928c20f2107ccc81f/plugins/hid/hiddmxdevice.h#L34
+const uint16_t DMXCProjectsNodleU1Factory::VENDOR_ID_FX5 = 0x16c0;
+const uint16_t DMXCProjectsNodleU1Factory::PRODUCT_ID_FX5 = 0x088b;
+
 bool DMXCProjectsNodleU1Factory::DeviceAdded(
     WidgetObserver *observer,
     libusb_device *usb_device,
     const struct libusb_device_descriptor &descriptor) {
-  if (descriptor.idVendor != VENDOR_ID || descriptor.idProduct != PRODUCT_ID) {
+  OLA_DEBUG << "Factory, DeviceAdded";
+  if (
+    ((descriptor.idVendor != VENDOR_ID) &&
+    (descriptor.idVendor != VENDOR_ID_FX5)) ||
+    ((descriptor.idProduct != PRODUCT_ID) &&
+    (descriptor.idProduct != PRODUCT_ID_FX5))
+    ) {
     return false;
   }
 
-  OLA_INFO << "Found a new Nodle U1 device";
+  OLA_INFO << "Found a new Nodle U1 or clone device";
   ola::usb::LibUsbAdaptor::DeviceInformation info;
   if (!m_adaptor->GetDeviceInfo(usb_device, descriptor, &info)) {
     return false;
   }
 
   OLA_INFO << "Nodle U1 serial: " << info.serial;
 
+  // Check if it's a RP2040-based widget and if so, how many ins and outs it has
+  int ret = 0;
+  unsigned int ins = 1;   // Input universes
+  unsigned int outs = 1;  // Output universes
+  char variant = 'S';     // Variant: S = simple, R = RDM (not yet implemented)
+  if (info.serial.find('RP2040_') != std::string::npos) {
+    // Model format: ??Tx ??Rx S
+    ret = sscanf(info.product.c_str(), "%uTx %uRx %c", &outs, &ins, &variant);
```

Comment:
Maybe the number of INs and OUTs should not be derived from the USB product string. We could make the modes switchable. This makes sense when the hardware around (bus drivers etc) is flexibly wired. If it's all fixed and soldered, this solution here might be better. I'm unsure about it ...